lib/Doctrine/DBAL/Platforms/DB2Platform.php

```diff
@@ -878,4 +881,57 @@ protected function getReservedKeywordsClass()
     {
         return Keywords\DB2Keywords::class;
     }
+    
+    /**
+     * Prepare ORDER BY string for OVER() if applicable
+     * 
+     * @param  string $query
+     * 
+     * @return string
+     */
+    private function getOrderByForOver($query)
+    {
+        //preset empty array
+        $orderByArray = [];
+
+        // remove extra white space
+        $query = preg_replace('/\s+/', ' ', $query);
+
+        //determine if 'ORDER BY' is part of the query
+        $orderByFound = preg_match('/(\bORDER\b)(\s*)(\bBY\b)(?!.*\b\1\b)/i', $query, $orderByPositionArray, PREG_OFFSET_CAPTURE);
+
+        // ORDER BY found in query string
+        if (0 !== $orderByFound) {
+            $orderByPosition = $orderByPositionArray[0][1];
+
+            $queryArray = preg_split('/[, ]/', substr($query, 0, $orderByPosition -1));
+            $orderByArray = explode(',', substr($query, $orderByPosition + strlen('ORDER BY')));
+
+            foreach ($orderByArray as $orderIndex => $orderValue) {
+                $splitOrder = explode(' ', $orderValue);
+
+                foreach ($splitOrder as $splitIndex => $splitValue) {
+                    switch (strtoupper($splitValue)) {
+                        case 'ASC':
+                        case 'DESC':
+                            break;
+                        default:
+                            $splitOrder[$splitIndex] = array_search($splitValue, $queryArray) === false
+                                                       ? $splitValue
+                                                       : $queryArray[array_search($splitValue, $queryArray)+2];
+                            break;
+                    }
+                }
+                $orderByArray[$orderIndex] = array_filter($splitOrder);
```

Comment:
What are the expected empty values in this case? Should `'0'` be filtered too? Looks like it can be fixed by a proper replacement of the `explode(' ')` above.

Reply:
not 100% sure yet, the filtered values are primarily "" (empty string).
Might have to investigate more...